### DISCUSSION TITLE
Add a beatmap to a collection from a playlist

### DISCUSSION THREAD

**Discussion by camellirite** [ppy/osu#19019] [Ideas] [answered]
I just played a fun map in the spotlights playlist and I wanted to add it to a collection so I didn't forget about it. I found out I couldn't right click the beatmaps in the playlist to add them

**Comment by Loreos7** [comment]
You must download a beatmap before adding to a collection.

I dont think playlist interface should contain the right click menu.

https://user-images.githubusercontent.com/86934170/177611085-103a3571-e68b-498b-ab7f-fe32b498e1f9.mp4

**Reply by frenzibyte** [reply]
Not sure about having to 1. download the beatmaps then 2. leave playlists then 3. look for each of them to add to collection then 4. come back to the playlists. I can see it pretty useful during multiplayer sessions.

**Comment by frenzibyte** [comment]
I think it would be better having it shown as a button next to the results screen button, but there's not proper UI for that yet so I went with a context menu instead:

https://user-images.githubusercontent.com/22781491/177635942-38e9854e-b4e7-4474-8f6a-b2f093f0ae00.mp4

Will wait for further feedback before I consider PR'ing this (cc @peppy).

**Reply by smoogipoo** [reply]
I prefer this over the results screen suggestion.

**Reply by frenzibyte** [reply]
Just to clarify, I meant a button next to the results screen button in the playlist item.

**Reply by smoogipoo** [reply]
Ah. Eh... I don't think it's that common of a thing to require a permanent button. This way is the same as stable, and I haven't heard complaints about it.

**Reply by peppy** [reply]
I think it's fine.

**Comment by frenzibyte** [comment, marked as answer]
Will be resolved in #19037
